File: backend/src/models/Item.js
const pool = require('../config/database');
const { v4: uuidv4 } = require('uuid');

const Item = {
  async create(wishlistId, name, description = '', price = null, imageUrl = null, sourceUrl = null) {
    const id = uuidv4();
    const createdAt = new Date();

    const result = await pool.query(
      'INSERT INTO items (id, wishlist_id, name, description, price, image_url, source_url, created_at) VALUES ($1, $2, $3, $4, $5, $6, $7, $8) RETURNING *',
      [id, wishlistId, name, description, price, imageUrl, sourceUrl, createdAt]
    );

    return result.rows[0];
  },

  async findById(id) {
    const result = await pool.query('SELECT * FROM items WHERE id = $1', [id]);
    return result.rows[0];
  },

  async findByWishlistId(wishlistId) {
    const result = await pool.query('SELECT * FROM items WHERE wishlist_id = $1 ORDER BY created_at DESC', [wishlistId]);
    return result.rows;
  },

  async update(id, name, description, price, imageUrl, sourceUrl) {
    const result = await pool.query(
      'UPDATE items SET name = $1, description = $2, price = $3, image_url = $4, source_url = $5 WHERE id = $6 RETURNING *',
      [name, description, price, imageUrl, sourceUrl, id]
    );
    return result.rows[0];
  },

  async delete(id) {
    await pool.query('DELETE FROM items WHERE id = $1', [id]);
  },

  async toggleCompleted(id, completed) {
    const result = await pool.query(
      'UPDATE items SET completed = $1, completed_at = $2 WHERE id = $3 RETURNING *',
      [completed, completed ? new Date() : null, id]
    );
    return result.rows[0];
  },
};

module.exports = Item;

File: backend/models/Item.py
from config.database import Database

class Item:
    """Item model for database operations"""
    
    @staticmethod
    def create(wishlist_id, name, description=None, price=None, image_url=None, source_url=None):
        """Create a new item"""
        sql = '''
            INSERT INTO items (wishlist_id, name, description, price, image_url, source_url)
            VALUES (%s, %s, %s, %s, %s, %s)
            RETURNING *
        '''
        return Database.execute_and_fetch(sql, (wishlist_id, name, description, price, image_url, source_url))
    
    @staticmethod
    def get_by_id(item_id):
        """Get item by ID"""
        sql = 'SELECT * FROM items WHERE id = %s'
        return Database.query_one(sql, (item_id,))
    
    @staticmethod
    def get_by_wishlist(wishlist_id):
        """Get all items in a wishlist"""
        sql = 'SELECT * FROM items WHERE wishlist_id = %s ORDER BY created_at DESC'
        return Database.query(sql, (wishlist_id,))
    
    @staticmethod
    def update(item_id, **kwargs):
        """Update item fields"""
        allowed_fields = ['name', 'description', 'price', 'image_url', 'source_url', 'completed', 'completed_at']
        updates = {k: v for k, v in kwargs.items() if k in allowed_fields}
        
        if not updates:
            return None
        
        set_clause = ', '.join([f'{k} = %s' for k in updates.keys()])
        values = list(updates.values()) + [item_id]
        
        sql = f'UPDATE items SET {set_clause} WHERE id = %s RETURNING *'
        return Database.execute_and_fetch(sql, values)
    
    @staticmethod
    def toggle_completion(item_id):
        """Toggle item completion status"""
        sql = '''
            UPDATE items 
            SET completed = NOT completed,
                completed_at = CASE 
                    WHEN completed = FALSE THEN CURRENT_TIMESTAMP
                    ELSE NULL
                END
            WHERE id = %s
            RETURNING *
        '''
        return Database.execute_and_fetch(sql, (item_id,))
    
    @staticmethod
    def delete(item_id):
        """Delete item"""
        sql = 'DELETE FROM items WHERE id = %s'
        return Database.execute(sql, (item_id,))
